src/axom/primal/operators/detail/evaluate_integral_impl.hpp

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2017-2022, Lawrence Livermore National Security, LLC and
+// other Axom Project Developers. See the top-level LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+
+#ifndef PRIMAL_EVAL_INTEGRAL_IMPL_HPP_
+#define PRIMAL_EVAL_INTEGRAL_IMPL_HPP_
+
+// Axom includes
+#include "axom/config.hpp"  // for compile-time configuration options
+#include "axom/primal.hpp"
+
+// MFEM includes
+#ifdef AXOM_USE_MFEM
+  #include "mfem.hpp"
+#else
+  #error "Primal's integral evaluation functions require mfem library."
+#endif
+
+// C++ includes
+#include <cmath>
+
+namespace axom
+{
+namespace primal
+{
+namespace detail
+{
+/*!
+ * \brief Evaluate a scalar field line integral on a single Bezier curve.
+ *
+ * Evaluate the scalar field line integral with MFEM integration rule
+ *
+ * \param [in] c the Bezier curve object
+ * \param [in] integrand the lambda function representing the integrand. 
+ * Must accept a 2D point as input and return a double
+ * \param [in] quad the mfem integration rule containing nodes and weights
+ * \return the value of the integral
+ */
+inline double evaluate_line_integral_component(
+  const primal::BezierCurve<double, 2>& c,
+  std::function<double(Point2D)> integrand,
```

Comment:
I think the lambda should be passed by r-value ref (here and elsewhere):

```suggestion
  std::function<double(Point2D)>&& integrand,
```

(@kanye-quest -- could you please confirm?)

Reply:
I'm very unclear on the details, but passing the `std::function`'s by R-value reference is causing some issues with some to-be-implemented winding number code. Specifically, if we pass to `evaluate_line_integral` a lambda function that is the return value for some other function (see below), then `evaluate_line_integral_component` can no longer fit the argument to either overloaded definition. This doesn't happen when the `std::function`'s are passed by constant reference, or when the lambda function is not the return value for another function, so I am unsure how to keep this as general as possible.

```C++
// An example of function that returns argument to evaluate_line_integral
std::function<Vector2D(Point2D)> get_vector_field(Point2D p) 
{
  return [p](Point2D x) -> Vector2D { return Vector2D({p[1], p[0]}); };
}
```

Reply:
Thanks -- ok, sounds like something to think about and try to fix in the future.

Reply:
This might have something to do with this function (and the one below) accepting an `std::function` instead of a `Lambda&&` like in the other methods. The latter is a universal reference, so it can bind to both lvalues and rvalues.

For what it's worth, I think accepting `Lambda&&` might be better from an inlineability perspective -- it would be much more difficult for the compiler to inline through a type-erased `std::function`.